doc/src/manual/calling-c-and-fortran-code.md

```diff
@@ -804,13 +778,9 @@ end
 ```
 
 Here, the input `p` is declared to be of type `Ref{gsl_permutation}`, meaning that the memory
-that `p` points to may be managed by Julia or by C. A pointer to memory allocated by C should
+that `p` points to may be managed by Julia or by C. A pointer to memory allocated by C should generally
 be of type `Ptr{gsl_permutation}`, but it is convertible using [`Base.cconvert`](@ref) and therefore
-can be used in the same (covariant) context of the input argument to a [`ccall`](@ref). A pointer
-to memory allocated by Julia must be of type `Ref{gsl_permutation}`, to ensure that the memory
-address pointed to is valid and that Julia's garbage collector manages the chunk of memory pointed
-to correctly. Therefore, the `Ref{gsl_permutation}` declaration allows pointers managed by C or
-Julia to be used.
+can be used in the same (covariant) context of the input argument to a [`ccall`](@ref).
 
 If the C wrapper never expects the user to pass pointers to memory managed by Julia, then using
```

Comment:
```suggestion
Now if you look closely enough at this example, you may notice that it is incorrect, given our explanation above of preferred declaration types. Do you see it? The function we are calling is going to free the memory. This type of operation cannot be given a Julia object (it will crash or cause memory corruption). Therefore, it may be preferable to declare the `p` type as `Ptr{gsl_permutation }`, to make it harder for the user to mistakenly pass another sort of object there than one obtained via `gsl_permutation_alloc`.

If the C wrapper never expects the user to pass pointers to memory managed by Julia, then using
```